storage/filesystem/dotgit/dotgit.go

```diff
@@ -780,33 +788,23 @@ func (d *DotGit) RemoveRef(name plumbing.ReferenceName) error {
 }
 
 func (d *DotGit) addRefsFromPackedRefs(refs *[]*plumbing.Reference, seen map[plumbing.ReferenceName]bool) (err error) {
-	packedRefs, err := d.findPackedRefs()
-	if err != nil {
-		return err
-	}
-
-	for _, ref := range packedRefs {
+	return d.findPackedRefs(func(ref *plumbing.Reference) bool {
 		if !seen[ref.Name()] {
 			*refs = append(*refs, ref)
 			seen[ref.Name()] = true
 		}
-	}
-	return nil
+		return true
+	})
 }
 
 func (d *DotGit) addRefsFromPackedRefsFile(refs *[]*plumbing.Reference, f billy.File, seen map[plumbing.ReferenceName]bool) (err error) {
-	packedRefs, err := d.findPackedRefsInFile(f)
-	if err != nil {
-		return err
-	}
-
-	for _, ref := range packedRefs {
+	return d.findPackedRefsInFile(f, func(ref *plumbing.Reference) bool {
 		if !seen[ref.Name()] {
```

Comment:
Although line `739` checks for `nil`, this func should also check for it. The reasoning being that future execution paths may assume it would be safe to call in with a nil `ref`.